services/bettersocial/post/blockAnonymousComment.js

```diff
@@ -0,0 +1,116 @@
+const UserBlockUserFunction = require('../../../databases/functions/userBlockUser');
+const UserFollowUserFunction = require('../../../databases/functions/userFollowUser');
+const UsersFunction = require('../../../databases/functions/users');
+const {
+  User,
+  UserFollowUser,
+  UserFollowUserHistory,
+  UserBlockedUser,
+  UserBlockedUserHistory,
+  sequelize
+} = require('../../../databases/models');
+const Getstream = require('../../../vendor/getstream');
+const RedisBlockHelper = require('../../redis/helper/RedisBlockHelper');
+const BetterSocialScoreBlockUser = require('../score/block-user');
+
+/**
+ * @typedef {Object} BetterSocialBlockAnonymousPostV2OptionalParams
+ * @property {Object[]} [reason]
+ * @property {string} [message]
+ */
+/**
+ *
+ * @param {string} token
+ * @param {string} selfUserId
+ * @param {string} postId
+ * @param {string} source
+ * @param {BetterSocialBlockAnonymousPostV2OptionalParams} params
+ * @returns
+ */
+const BetterSocialBlockAnonymousComment = async (
+  res,
+  token,
+  selfUserId,
+  postId,
+  commentId,
+  source,
+  params = {}
+) => {
+  let authorAnonymousUserId = null;
+  const {reason = [], message = ''} = params;
+  const selfAnonymousUserId = await UsersFunction.findAnonymousUserId(User, selfUserId);
+
+  try {
+    authorAnonymousUserId = await Getstream.feed.getUserIdFromSource(res, source, {
+      post_id: postId,
+      comment_id: commentId
+    });
+  } catch (e) {
+    return {
+      isSuccess: false,
+      message: e?.message || 'Error in fetching getstream post by post id'
+    };
+  }
+  console.log(`${selfAnonymousUserId?.user_id} vs ${authorAnonymousUserId}`);
+  if (selfAnonymousUserId?.user_id === authorAnonymousUserId)
+    return {
+      isSuccess: false,
+      message: "You can't block your own post"
+    };
+
+  try {
+    await sequelize.transaction(async (t) => {
+      await UserFollowUserFunction.userBlock(
+        UserFollowUser,
+        UserFollowUserHistory,
+        authorAnonymousUserId,
+        selfUserId,
+        {transaction: t, postId, commentId}
+      );
+
+      await UserBlockUserFunction.userBlock(
+        UserBlockedUser,
+        UserBlockedUserHistory,
+        selfUserId,
+        authorAnonymousUserId,
+        source,
+        {transaction: t, message, postId, reason, isAnonymous: true}
+      );
+    });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This transaction block lacks error handling. If any of the operations inside the transaction fail, the transaction will be rolled back but no specific error message will be returned. Consider adding error handling inside the transaction to provide more detailed feedback.



<!-- This is an auto-generated comment by OSS CodeRabbit -->